### PR TITLE
Put completeness in product meta, to avoid a fetch after product loading

### PR DESCRIPTION
**Pull request by momoss**
Put completeness in product meta, to avoid a fetch after product loading

| Q                                 | A
| --------------------------------- | ---
| Added Specs                       | -
| Added Behats                      | -
| Added integration tests           | -
| Changelog updated                 | -
| Review and 2 GTM                  | Ok
| Micro Demo to the PO (Story only) | -
| Migration script                  | -
| Tech Doc                          | -

`Todo`: Pending / Work in progress
`OK`: Done / Validated
`-`: Not needed

### CODE REVIEW

File: src/Pim/Bundle/EnrichBundle/Resources/public/js/product/form/attributes/completeness.js [hunk 1]
```diff
@@ -30,10 +30,10 @@ define(
              * @param {object} event
              */
             addFieldFilter: function (event) {
-                event.filters.push(fetcherRegistry.getFetcher('product-completeness').fetchForProduct(
-                    this.getFormData().meta.id,
-                    this.getFormData().family
-                ).then(function (completenesses) {
+                event.filters.push($.Deferred().resolve({
```
Comment (damien-carcel) [inline src/Pim/Bundle/EnrichBundle/Resources/public/js/product/form/attributes/completeness.js]: As you did this, is the product-completeness fetcher still needed? For the widget on the main screen, maybe?

Comment (momoss) [inline src/Pim/Bundle/EnrichBundle/Resources/public/js/product/form/attributes/completeness.js]: Yes he is needed for completeness panel

File: src/Pim/Bundle/EnrichBundle/Controller/Rest/ProductController.php [hunk 2]
```diff
@@ -376,4 +422,29 @@ protected function updateProduct(ProductInterface $product, array $data)
...
     }
+
+    /**
+     * Get Product Completeness and normalize it
+     *
+     * @param ProductInterface $product
+     *
+     * @return array|scalar
+     */
+    protected function getNormalizeCompletenesses(ProductInterface $product)
```
Comment (phaseinducer) [inline src/Pim/Bundle/EnrichBundle/Controller/Rest/ProductController.php]: *normalized

File: src/Pim/Bundle/EnrichBundle/Controller/Rest/ProductController.php [hunk 3]
```diff
@@ -376,4 +422,29 @@ protected function updateProduct(ProductInterface $product, array $data)
 
         $this->productUpdater->update($product, $data);
     }
+
+    /**
+     * Get Product Completeness and normalize it
+     *
+     * @param ProductInterface $product
+     *
+     * @return array|scalar
```
Comment (phaseinducer) [inline src/Pim/Bundle/EnrichBundle/Controller/Rest/ProductController.php]: Why scalar?

File: src/Pim/Bundle/EnrichBundle/Controller/Rest/ProductController.php [hunk 4]
```diff
@@ -160,11 +200,14 @@ public function getAction($id)
             'disable_grouping_separator' => true
         ];
 
-        return new JsonResponse($this->normalizer->normalize(
+        $normalizeProduct = $this->normalizer->normalize(
```
Comment (phaseinducer) [inline src/Pim/Bundle/EnrichBundle/Controller/Rest/ProductController.php]: *normalized

File: src/Pim/Bundle/EnrichBundle/Controller/Rest/ProductController.php [hunk 5]
```diff
@@ -238,11 +281,14 @@ public function postAction(Request $request, $id)
                 'disable_grouping_separator' => true
             ];
 
-            return new JsonResponse($this->normalizer->normalize(
+            $normalizeProduct = $this->normalizer->normalize(
```
Comment (phaseinducer) [inline src/Pim/Bundle/EnrichBundle/Controller/Rest/ProductController.php]: idem

File: src/Pim/Bundle/EnrichBundle/Resources/public/js/product/form/attributes/completeness.js (hunk 1 above)
Comment (phaseinducer) [inline src/Pim/Bundle/EnrichBundle/Resources/public/js/product/form/attributes/completeness.js]: Actually the panel could use what you added and we could drop the fetcher imho.

**Pull request merged**